### PR TITLE
[TIR][Schedule] Support annotate dict typed value

### PR DESCRIPTION
**Pull request by wrongtest-intellif**
Allow  `s.annotate(loop_rv, pragma_key, {"cfg0":  1, "cfg1": 2, ...})` .

### CODE REVIEW

**Comment by junrushao1994** [comment]
CC: @vinx13

File: python/tvm/tir/schedule/_type_checker.py [hunk 1]
```diff
@@ -177,6 +197,19 @@ def _type_check_list(v: List[Any], name: str, type_: Any) -> Optional[str]:
...
+            return _type_check_err(dict_obj, name, dict)
+        for k, v in dict_obj.items():
+            error_msg = _type_check(k, f"{name}[{k}]", ktype_)
+            if error_msg is not None:
+                return error_msg
+            error_msg = _type_check(v, f"{name}[{k}]", vtype_)
+            if error_msg is not None:
+                return error_msg
+        return None
+
```
Comment (vinx13) [inline python/tvm/tir/schedule/_type_checker.py:212]: need test case in https://github.com/apache/tvm/blob/main/tests/python/unittest/test_type_annotation_checker.py

File: tests/python/unittest/test_type_annotation_checker.py [hunk 2]
```diff
@@ -16,7 +16,8 @@
 # under the License.
 """Test type checker based on python's type annotations"""
 
-from typing import List, Tuple, Union
+import sys
```
Comment (vinx13) [inline tests/python/unittest/test_type_annotation_checker.py:19]: this is not used

Comment (wrongtest-intellif) [reply tests/python/unittest/test_type_annotation_checker.py:19]:
```python
if __name__ == "__main__":
    sys.exit(pytest.main(sys.argv))
```

Comment (vinx13) [reply tests/python/unittest/test_type_annotation_checker.py:19]: ah didn't see this usage

**Review by vinx13 — APPROVED** [review]

**Pull request merged**